src/Rabbitmq/Manager/Manager.php

```diff
@@ -167,12 +186,20 @@ public function addCommands(array $commands, string $connectionIdentifier, array
 
         // send commands
         $channel = $connection->getChannel();
+
+        // establish publish confirm mode if necessary
+        $publishConfirmTimeout > 0 and $channel->confirm_select();
+
         $applicationHeaders = \Maleficarum\Ioc\Container::get('PhpAmqpLib\Wire\AMQPTable', [$commandsHeaders]);
         foreach ($commands as $command) {
             $message = \Maleficarum\Ioc\Container::get('PhpAmqpLib\Message\AMQPMessage', [$command->toJSON(), ['delivery_mode' => 2, 'application_headers' => $applicationHeaders]]);
             $channel->batch_basic_publish($message, $connection->getExchangeName(), $connection->getQueueName());
         }
         $channel->publish_batch();
+
+        // wait for publish confirm
+        $channel->wait_for_pending_acks($publishConfirmTimeout);
+
```

Comment:
If the confirm_select() is conditional why isn't wait_for_pending_acks() conditional as well?